vpr/src/pack/re_cluster_util.h

```diff
@@ -32,52 +33,73 @@ ClusterBlockId atom_to_cluster(const AtomBlockId& atom);
 std::vector<AtomBlockId> cluster_to_atoms(const ClusterBlockId& cluster);
 
 /**
- * @brief A function that loads the router data for a cluster
+ * @brief A function that loads the intra-cluster router data of one cluster
  */
 t_lb_router_data* lb_load_router_data(std::vector<t_lb_type_rr_node>* lb_type_rr_graphs,
-                                      const ClusterBlockId& clb_index);
+                                      const ClusterBlockId& clb_index,
+                                      const std::vector<AtomBlockId>& clb_atoms);
 
 /**
- * @brief A function that removes an atom from a cluster and check legality of
+ * @brief A function that removes a molecule from a cluster and check legality of
  * the old cluster. 
  *
  * It returns true if the removal is done and the old cluster is legal.
  * It aborts the removal and returns false if the removal will make the old cluster 
  * illegal
  */
-bool remove_atom_from_cluster(const AtomBlockId& atom_id,
-                              std::vector<t_lb_type_rr_node>* lb_type_rr_graphs,
-                              ClusterBlockId& old_clb,
-                              t_clustering_data& clustering_data,
-                              int& imacro,
-                              bool during_packing);
+bool remove_mol_from_cluster(const t_pack_molecule* molecule,
+                             int molecule_size,
+                             ClusterBlockId& old_clb,
+                             t_lb_router_data*& router_data);
 
 /**
  * @brief A function that starts a new cluster for one specific molecule
  * 
- * It place the molecule in a specific type and mode that should be passed by
+ * It places the molecule in a specific type and mode that should be passed by
  * the higher level routine.
```

Comment:
Goo d to describe what all the inputs do and what the output is.  clb_index seems to be an input that is supposed to be the index of the next clb to be created?  If so, could it instead  just be computed inside the start_new_cluster_for_mol function?
Describe during_packing.
clustering_data, router_data and temp_cluster_pr have to be in a specific state before calling and are updated by the routine?